aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -919,6 +919,41 @@ Tensor _nested_view_from_buffer(
     std::vector<int64_t>(offsets.begin(), offsets.end()));
 }
 
+/**
+ * Create a [B, *, D] nested tensor that is a view of a 2D values tensor
+ * where offsets are the cumsum of the size along the ragged dimension
+ *
+ * @return Returns a nested tensor that
+ * aliases the same storage as values
+ */
+Tensor _nested_from_values_and_offsets(
+    const Tensor& values,
+    const Tensor& offsets) {
+  TORCH_CHECK(values.dim() == 2, "values must be a 2D tensor, got values.dim() ", values.dim(), ".");
+  TORCH_CHECK(values.is_contiguous(), "values must be contiguous.");
+  TORCH_CHECK(offsets.dim() == 1, "offsets must be a 1D tensor, got offsets.dim() ", offsets.dim(), ".");
+
+  TORCH_CHECK(offsets.select(0, 0).item<int64_t>() == 0,
+    "First element of offsets must be equal to 0, got ",
+    offsets.select(0, 0).item<int64_t>(),
+    ".");
+  TORCH_CHECK(offsets.select(0, -1).item<int64_t>() == values.size(0),
```

Comment:
for `torch.compile()` purposes, we may not want to do these checks with item calls. There might be more done here that is incompatible with symbolically propagating shapes, TBD but I'll look into it

Reply:
That makes sense, I will make this private function an unsafe constructor for now then